### DISCUSSION TITLE
Aktueller Projektstatus & V2

### DISCUSSION THREAD

**Discussion by brutella** [brutella/hkknx-public#105] [Announcements]
# Aktueller Projektstatus und Ausblick auf Version 2.0

*tl;dr Version 1(.3) ist super. Version 2 wir noch besser mit Skripten → https://hochgatterer.me/de/hkknx/docs/scripts/*

Die HomeKit Brücke für KNX ist seit mehr als einem Jahr verfügbar. Mein ursprüngliches Ziel, eine HomeKit Brücke zur Verfügung zu stellen, die einfach zu benutzen ist, wurde auf jedem Fall erfüllt. Mit Version 1.3 wurden auch noch die letzten ausstehenden Funktionen hinzugefügt. Somit ist es nun möglich über HomeKit nicht nur Lampen, Rollladen und Garagentore zu steuern, sondern auch Szenen auszuführen, Alarmanlagen abzubilden und Lüftungsanlagen zu steuern.

Das bisherige Feedback der User ist durchgehend positiv und das macht mich stolz.

**Wie geht es nun weiter?**

Als langjähriger Entwickler für HomeKit kenne ich die Stärken und Schwächen von HomeKit ganz genau. Leider ist auch hier nicht alles Gold was glänzt. 

Die Bedienung der Geräte über iPhone, iPad und Siri ist zwar top, aber Automatisierungen sind mit HomeKit noch nicht zuverlässig möglich. Nach fast jedem iOS Update beklagen User, dass bestehende Automatisierungen nicht richtig ausgeführt werden. Oftmals hat es damit zu tun, dass eines der Apple-Geräte eine Beta-Version installiert hat, nicht mit dem Netzwerk verbunden, oder mit der Apple-ID angemeldet ist.

Man muss wissen, dass die HomeKit Automatisierungen nicht vom iPhone sondern von der Steuerzentrale ausgeführt werden. Wenn also eine Automatisierung erstellt oder geändert wird, müssen die Daten erstmals mit der Steuerzentrale synchronisiert werden. Das geschieht im Hintergrund ganz automatisch oder eine Möglichkeit den Fortschritt zu sehen. Falls also die Steuerzentrale nicht korrekt verbunden ist (weil die Internetverbindung nicht verfügbar ist, oder es ein Problem mit der Apple-ID gibt), dass funktionieren die Automatisierungen einfach nicht.

In Smart-Homes, in dem essenzielle Funktionen automatisiert werden, müssen Automatisierungen jedoch zuverlässig ausgeführt werden. In KNX kommen dafür Logik-Module zum Einsatz, aber komplexerer Abläufe sind damit nicht möglich. Oftmals werden auch dedizierte Logik-Server eingesetzt.

**Was hat das mit der HomeKit Brücke für KNX zu tun?**

Die HomeKit Brücke für KNX war von Anfang an als Alternative zu bestehenden Lösungen konzipiert – und dazu zählen auch Automatisierungen. Ziel ist es also, dass die HomeKit Brücke für KNX alle Anforderungen an ein modernes KNX-System erfüllt. Das beinhaltet Logiken programmieren, den KNX-Bus überwachen, bis hin zu Dashboards erstellen. Das Alles wird mit Hilfe von Skripten möglich sein.

Was genau Skripte sind und wie diese erstellt werden können, steht im [Online Handbuch](https://hochgatterer.me/de/hkknx/docs/scripts/).

**Wie ist der Status von Version 2?**

Seit einigen Monate arbeite ich bereits an Version 2.0 und die ersten Beta-Tester haben die neue Version bereits am Laufen.
Die Beta-Version ist zurzeit nur über die Webseite verfügbar. Die erste Pre-Release Version wird man auch bald über die Weboberfläche installieren können.
https://hochgatterer.me/de/hkknx/docs/scripts/

**Muss die Version 2 erneuert gekauft werden?**

Nein, Version 2 ist ein kostenloses Update für bestehende User. Eine bestehende Lizenz funktioniert also auch mit Version 2.
Sobald Version 2 offiziell verfügbar ist, wird es aber eine Preiserhöhung für die Lizenz geben.

Für das Versenden von Mitteilungen sowie das Abrufen von Wetterdaten wurde ein eigener Internetdienst entwickelt. Um diese Funktionen in Skripten verwenden zu können, ist ein Abonnement erforderlich. Mit Hilfe dieser Einnahmen werden die Hosting-Kosten des Servers gedeckt. Während der Testphase ist kein Abonnement erforderlich. Mehr Details dazu folgen noch.

**Comment by BugRoger** [comment]
Danke für das Update. Die Skriptfunktionalität is Hammer!!! 😄

**Comment by code-constructor** [comment]
Klingt sehr spannend. Habe die Software seit 2 Wochen im Einsatz und bin begeistert wie solide und durchdacht alles funktioniert. Danke! 

Ich freue mich auf das Update und auf die kommenden Möglichkeiten des Skriptens :+1:
